plugins/services/src/js/stores/SDKEndpointStore.js

```diff
@@ -0,0 +1,161 @@
+import PluginSDK from "PluginSDK";
+
+import { SERVER_ACTION } from "#SRC/js/constants/ActionTypes";
+import AppDispatcher from "#SRC/js/events/AppDispatcher";
+import GetSetBaseStore from "#SRC/js/stores/GetSetBaseStore";
+
+import {
+  REQUEST_SDK_ENDPOINTS_SUCCESS,
+  REQUEST_SDK_ENDPOINTS_ERROR,
+  REQUEST_SDK_ENDPOINT_SUCCESS,
+  REQUEST_SDK_ENDPOINT_ERROR
+} from "../constants/ActionTypes";
+
+import SDKEndpointActions from "../events/SDKEndpointActions";
+import SDKServiceEndpoint from "../structs/SDKServiceEndpoint";
+
+class SDKEndpointStore extends GetSetBaseStore {
+  constructor() {
+    super(...arguments);
+
+    const self = this;
+
+    this.getSet_data = {
+      services: {}
+    };
+
+    PluginSDK.addStoreConfig({
+      store: this,
+      storeID: this.storeID,
+      listenAlways: true
+    });
+
+    this.dispatcherIndex = AppDispatcher.register(function(payload) {
+      if (payload.source !== SERVER_ACTION) {
+        return false;
+      }
+
+      const { type, data } = payload.action;
+      switch (type) {
+        case REQUEST_SDK_ENDPOINTS_SUCCESS:
+          self.processNewEndpoints(data.serviceId, data.endpoints);
+          break;
+        case REQUEST_SDK_ENDPOINTS_ERROR:
+          self.setServiceEndpoints(data.serviceId, {
+            endpoints: [],
+            error: data.error.description
+          });
+          break;
+        case REQUEST_SDK_ENDPOINT_SUCCESS:
+          self.processNewEndpoint(
+            data.serviceId,
+            data.endpointName,
+            data.endpointData,
+            data.contentType
+          );
+          break;
+        case REQUEST_SDK_ENDPOINT_ERROR:
+          self.setServiceEndpoints(data.serviceId, {
+            endpoints: [],
+            error: data.error
+          });
+          break;
+      }
+
+      return true;
+    });
+  }
+
+  getEndpointServices() {
+    const services = Object.assign({}, this.get("services"));
+
+    Object.keys(services).forEach(serviceId => {
```

Comment:
We'll have to either clone this in various other spots when needed or use immutable.js for that. I think it is not necessary to use a new library now and it would be a bit messy to clone elsewhere, so I used our deepCopy utility to keep it clean.